File: 21 - Apriori/R/apriori.R
#Apriori
#install.packages('arules')
library(arules)
dataset = read.csv('Data/Market_Basket_Optimisation.csv', header = FALSE)
dataset = read.transactions('Data/Market_Basket_Optimisation.csv', sep = ',', rm.duplicates = TRUE)
summary(dataset)
itemFrequencyPlot(dataset, topN = 10)

#Training Apriori on the dataset
rules = apriori(data = dataset,
                parameter = list(support =0.004 , confidence = 0.20))


#Visualising the results
inspect(sort(rules,by = 'lift')[1:10])

File: 21 - Apriori/Python/apriori.py
# -*- coding: utf-8 -*-
"""
Created on Fri Jul 26 02:19:18 2019

@author: Thiago Montenegro
"""

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

dataset = pd.read_csv('Data/Market_Basket_Optimisation.csv', header = None)

transactions = []

for i in range(0, len(dataset)):
    transactions.append([str(dataset.values[i,j]) for j in range(0,20)])

#Training Apriori On the dataset
from apyori import apriori
rules = apriori(transactions, min_support = 0.003 , min_confidence = 0.4, min_lift =3 , min_length = 2)
#Support calculus
#products whichj is purchased 3 products by day 
# 3 * 7 / total number_of_transactions


#Visualising the transactions
results = list(rules)
